Runtime/Model/Metrics/MetricsSubmissionQueue.cs

```diff
@@ -112,12 +112,15 @@ internal void SendPayload(ICollection<T> events, uint attempts = 0)
 
             RequestHandler.Post(SubmissionUrl, payload, (long statusCode, bool httpError, string response) =>
             {
-                if (statusCode == 200)
+                if (httpError == false && statusCode == 200)
                 {
                     OnRequestCompleted();
                 }
-                else if (statusCode > 501 && statusCode != 505)
+                else
```

Comment:
I'm not sure about that. Those statuses were confirmed with the team and Paul when we were working on the integration. In the situation when status code = 403/404 it doesn't make sense to retry.

Reply:
Ok. What about when `httpError == true`? In this case, the metric is never sent. This can happen if the connection is temporarily down for instance.

Reply:
how about extending the check by checking if `httpError == true || (statusCode > 501 && statusCode != 505)` ?